torch/csrc/jit/script/compiler.cpp

```diff
@@ -319,7 +293,9 @@ struct Environment {
       const SourceRange& loc,
       const std::string& name,
       SugaredValuePtr value) {
+    WithInsertPoint guard(b->return_node());
```

Comment:
Huh? I think this is very surprising behavior. It also makes an implicit assumption that we never jump to the middle of the block and emit something. It would be better to make sure we are calling setSugaredVar while we are still in the current environment.